jib-core/src/main/java/com/google/cloud/tools/jib/builder/steps/ObtainBaseImageLayerStep.java

```diff
@@ -36,14 +38,49 @@
 import javax.annotation.Nullable;
 
 /** Pulls and caches a single base image layer. */
-class PullAndCacheBaseImageLayerStep implements Callable<CachedLayerAndName> {
+class ObtainBaseImageLayerStep implements Callable<PreparedLayer> {
 
   private static final String DESCRIPTION = "Pulling base image layer %s";
 
-  static ImmutableList<PullAndCacheBaseImageLayerStep> makeList(
+  @FunctionalInterface
+  private interface BlobExistenceChecker {
+
+    Optional<Boolean> exists(DescriptorDigest digest) throws IOException, RegistryException;
+  }
+
+  static ImmutableList<ObtainBaseImageLayerStep> makeListForForcedDownload(
       BuildConfiguration buildConfiguration,
       ProgressEventDispatcher.Factory progressEventDispatcherFactory,
       ImageAndAuthorization baseImageAndAuth) {
+    BlobExistenceChecker noOpBlobChecker = ignored -> Optional.empty();
+    return makeList(
+        buildConfiguration, progressEventDispatcherFactory, baseImageAndAuth, noOpBlobChecker);
+  }
+
+  static ImmutableList<ObtainBaseImageLayerStep> makeListForSelectiveDownload(
+      BuildConfiguration buildConfiguration,
+      ProgressEventDispatcher.Factory progressEventDispatcherFactory,
+      ImageAndAuthorization baseImageAndAuth,
+      Authorization pushAuthorization) {
+    Verify.verify(!buildConfiguration.isOffline());
+
+    RegistryClient targetRegistryClient =
+        buildConfiguration
+            .newTargetImageRegistryClientFactory()
+            .setAuthorization(pushAuthorization)
+            .newRegistryClient();
+    BlobExistenceChecker blobExistenceChecker =
+        digest -> Optional.of(targetRegistryClient.checkBlob(digest).isPresent());
```

Comment:
Hmm, this doesn't account for the cross-repo blob-mount case.  Maybe add a TODO for now.